### PR TITLE
API-17394 Remove eventing system

### PR DESCRIPTION
**Pull request by ElleeB**
For initial review

I will be breaking this up - it's huge.

A separate PR will also be created for the drop event subscription migration.

### CODE REVIEW

File: modules/appeals_api/config/schemas/v1/10182_headers.json [hunk 1]
```diff
@@ -15,10 +15,6 @@
         "X-VA-SSN":             { "$ref": "#/definitions/X-VA-SSN" },
         "X-VA-File-Number":     { "$ref": "#/definitions/X-VA-File-Number" },
         "X-VA-Birth-Date":      { "$ref": "#/definitions/X-VA-Birth-Date" },
-        "X-VA-Claimant-First-Name":     { "$ref": "#/definitions/X-VA-Claimant-First-Name" },
```
Comment (ElleeB) [inline modules/appeals_api/config/schemas/v1/10182_headers.json:18]: A point of confusion - claimant headers still remain in nod v1 header schema, but no claimant data in form schema.

File: modules/appeals_api/spec/fixtures/v1/valid_10182_headers.json [hunk 2]
```diff
@@ -5,10 +5,6 @@
   "X-VA-SSN": "123456789",
   "X-VA-File-Number": "987654321",
   "X-VA-Birth-Date": "1969-12-31",
-  "X-VA-Claimant-First-Name": "John",
```
Comment (ElleeB) [inline modules/appeals_api/spec/fixtures/v1/valid_10182_headers.json:8]: See comment above.

File: modules/appeals_api/spec/models/higher_level_review_spec.rb [hunk 3]
```diff
@@ -221,82 +221,64 @@
...
-      handler = instance_double(AppealsApi::Events::Handler)
-      allow(AppealsApi::Events::Handler).to receive(:new).and_return(handler)
-      allow(handler).to receive(:handle!)
-
-      higher_level_review.update_status!(status: 'pending')
-
-      expect(handler).to have_received(:handle!)
+    context 'when incoming and current statuses are different' do
+      it 'enqueues the status updated job' do
+        assert_equal 0, AppealsApi::AppealReceivedJob.jobs.size
```
Comment (cilestin) [inline modules/appeals_api/spec/models/higher_level_review_spec.rb:226]: `assert_equal` is a different testing framework (Rails' TestUnit) than rspec. Please update to use `expect` syntax.

Comment (cilestin) [reply modules/appeals_api/spec/models/higher_level_review_spec.rb:226]: Also, shouldn't this be making sure that `StatusUpdatedJob`'s size changed, not `AppealReceivedJob`? I'd personally add in "extra" expectations around both types of jobs to ensure they're both in the state you expect, since none/either/both can be created given different situations.

File: modules/appeals_api/spec/models/supplemental_claim_spec.rb [hunk 4]
```diff
@@ -155,98 +155,64 @@
...
-                                              'date_submitted' =>
-                                                sc_with_nvc.created_at.in_time_zone('America/Detroit').iso8601,
-                                              'guid' => sc_with_nvc.id,
-                                              'claimant_email' => 'joe@email.com',
-                                              'claimant_first_name' => 'joe'
-                                            }
-                                          ])
+    context 'when incoming and current statuses are different' do
+      it 'enqueues the status updated job' do
+        assert_equal 0, AppealsApi::AppealReceivedJob.jobs.size
```
Comment (cilestin) [inline modules/appeals_api/spec/models/supplemental_claim_spec.rb:160]: Should be checking for StatusUpdatedJob

File: modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb [hunk 5]
```diff
@@ -61,7 +61,7 @@ def new_base_path(path)
 
     context 'when a required headers is missing' do
       it 'returns an error' do
-        post(path, params: @data, headers: @headers.except('X-VA-File-Number'))
+        post(path, params: @minimum_data, headers: @headers.except('X-VA-File-Number'))
```
Comment (cilestin) [inline modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb:64]: Why these changes to use `@minimum_data`? It doesn't seem to make a material difference to the test itself - was it just easier to debug with smaller data?

**Review by cilestin — CHANGES_REQUESTED** [review]
A good first pass. Added some comments on things that stuck out but will probably go through with a finer comb when you split up into more easily digestible PRs.

I think this could be a good way to split off the PRs. Each can be rolled up into the previous if small enough:
1. Add the new jobs
2. Add specs for the new jobs
3. Update update_status! to use new jobs, update model specs
--- we should probably pause here and deploy, so we can easily roll back if any issues show up. If all is well, then...-----
5. Remove AppealsApi::Event workers & specs
6. Remove Event Subscription, table(s), & specs
7. Anything I forgot

File: modules/appeals_api/app/models/appeals_api/higher_level_review.rb [hunk 6]
```diff
@@ -344,8 +329,42 @@ def address_combined
...
+          from: current_status,
+          to: status.to_s,
+          status_update_time: Time.zone.now.iso8601,
+          statusable_id: id
+        }
+      )
+
+      return if auth_headers.blank? # Go no further if we've removed PII
+
+      create_appeal_received_job if status == 'submitted' && email_present?
```
[the remark this answers is not in the document]
Comment (ElleeB) [reply modules/appeals_api/app/models/appeals_api/higher_level_review.rb:353]: This was a tangle. I had them separate, both called independently of one another in the update_status method. But, actually, as I'm typing I think i have a solution to go back to. 😜

File: modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb [hunk 7]
```diff
@@ -225,33 +226,17 @@ def stamp_text
     end
 
     def update_status!(status:, code: nil, detail: nil)
-      current_status = self.status
-      update!(status: status, code: code, detail: detail)
+      @current_status ||= self.status
```
[the remark this answers is not in the document]
Comment (ElleeB) [reply modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb:229]: Because derp. Thanks!

File: modules/appeals_api/spec/models/higher_level_review_spec.rb (hunk 3 above)
Comment (ElleeB) [reply modules/appeals_api/spec/models/higher_level_review_spec.rb:226]: The calling of the same job over and over is another drafty derp. As for `assert_equal`, I'll go back to expect, but there were some strange side effects I was trying to avoid and this resolved it. I got excited after researching sidekiq testing. not gonna like.

File: modules/appeals_api/spec/models/higher_level_review_spec.rb [hunk 8]
```diff
@@ -221,82 +221,64 @@
...
-      handler = instance_double(AppealsApi::Events::Handler)
-      allow(AppealsApi::Events::Handler).to receive(:new).and_return(handler)
-      allow(handler).to receive(:handle!)
-
-      higher_level_review.update_status!(status: higher_level_review.status)
-
-      expect(handler).not_to have_received(:handle!)
+    context 'when incoming and current statuses are the same' do
+      it 'does not enqueues the status updated job' do
+        assert_equal 0, AppealsApi::AppealReceivedJob.jobs.size
```
[the remark this answers is not in the document]
Comment (ElleeB) [reply modules/appeals_api/spec/models/higher_level_review_spec.rb:234]: I cut and pasted my code and just didn't change the job name - just doing all the good work.

File: modules/appeals_api/spec/models/supplemental_claim_spec.rb (hunk 4 above)
Comment (ElleeB) [reply modules/appeals_api/spec/models/supplemental_claim_spec.rb:160]: I'm on it 😆

File: modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb (hunk 5 above)
Comment (ElleeB) [reply modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb:64]: Because there was never an @data to begin with 😬 🤷‍♀️

**Comment by cilestin** [comment]
Closing since this work was done by other PRs.
